website_and_docs/content/documentation/webdriver/troubleshooting/logging.en.md

```diff
@@ -9,70 +9,165 @@ description: >
+<pre>
+  Selenium::WebDriver.logger.ignore(%i[jwp_caps pause pauses])
+</pre>
+<br>
+<p>To turn off all <i>deprecation</i> notices:</p>
+<pre>
+  Selenium::WebDriver.logger.ignore(:deprecations)
+</pre>
+<br>
+<p><b>INFO</b></p>
+
+<p>This is where the most useful information gets logged. Selenium logs the endpoints and payloads
 sent to and received from the driver or server. This is a great way to see what Selenium is actually
 doing under the hood, and can be used to determine if it is Selenium code or driver code that
 is causing a problem. (Unfortunately, we can't blame the driver if Selenium is sending incorrect syntax).
+</p>
+
+<p><b>DEBUG</b></p>
+
+<p>This is less useful information where we log things about the servers and the sockets, and header information, etc.
+Debug mode is set if either <i>`$DEBUG`</i> is true or <i>`ENV['DEBUG']`</i> has a value.</p>
+{{< /tab >}}
+
+{{< tab header="JavaScript" >}}
+<p>JS has implemented logging APIs. It has also added LogManager to maintain a collection of loggers.</p>
+<p><b>1. Getting a logger can be done in two ways:</b></p>
+<pre>
+  const logging = require('selenium-webdriver/lib/logging')
+</pre>
+<br>
+<p>First method using manager</p>
+<pre>
+  mgr = new logging.LogManager()
+  logger = mgr.getLogger('')
+</pre>
+<br>
+<p>Second method without manager</p>
+<pre>
+  logger = logging.getLogger('')
+</pre>
+<br>
+<p>Loggers class uses hierarchical, dot-delimited namespaces (e.g. "" > "webdriver" > "webdriver.logging"). For example, the following code will give a hierarchy of loggers:</p>
+<pre>
+  logger = mgr.getLogger('foo.bar.baz')
+</pre>
+<br>
+<p>"" (parent of) > "foo" (parent of) > "foo.bar" (parent of) > "foo.bar.baz" (=logger)</p>
+
+<p><b>2. Logger Level:</b></p>
+
+<p>The basic levels for JS loggers are: <i>`OFF`, `SEVERE`, `WARNING`, `INFO`, `DEBUG`, `FINE`, `FINER`, `FINEST`, `ALL`</i>.<br>
+To log messages at the given level pass the level in the <i>`log()`</i> method:</p>
+<pre>
+  logger.log(logging.Level.INFO, 'This is an info message')
+</pre>
+<br>
+<p>You can also use the instance methods to target logs of a particular level:<p>
+<pre>
+  logger.finest('this is the finest message')
+  logger.finer('this is a finer message')
+  logger.info('this is an info message')
+  logger.warning('this is a warning message')
+  logger.severe('this is a severe message')
+</pre>
+<br>
+<p>By default the logger's level is <i>`Level.OFF`.</i> To change the level of the logger:</p>
+<pre>
+  logger.setLevel(logging.Level.INFO)
+</pre>
+<br>
+<p><b>3. Log Output:</b></p>
+
+<p>As opposed to ruby which by default sends logs to console in stdout, JS has the option to add a handler. The handler will be invoked for each message logged with this instance, or any of its descendants.</p>
+<pre>
+  logger.addHandler(callback)
+</pre>
+<br>
+<p>JS has provided a method to add console handler to the given logger. The console handler will log all messages using the JavaScript Console API:</p>
```

Comment:
It is probably an issue of me not understanding JavaScript, but what is the difference between a handler and a console handler?

Reply:
A handler will take a callback function and it will be invoked when the event(log) occurs. Console handler is a special type of handler which will display the logs on console in a fixed format. 
https://github.com/SeleniumHQ/selenium/blob/trunk/javascript/node/selenium-webdriver/lib/logging.js#L526-L559

But for user's perspective to display selenium logs, we will use `installConsoleHandler()`.